### DISCUSSION TITLE
How do I disable ALE spamming comments into my code

### DISCUSSION THREAD

**Discussion by klausman** [w0rp/ale#4405] [General]
I personally find that super disruptive, especially since I am using vim, and the gutter indicator (and status line info when I go there) are much less disruptive and easier to use when coding (as opposed to editing). The added comments are a storm of info I don't care about most of the time (when coding, there are a lot fewer moments where the code is fine than there are moments when e.g. a loop is half-written).

I tried finding an option to go to the old behavior (marker and status line only), but to no avail.

**Comment by RyanSquared** [comment]
do you mean the `g:ale_virtualtext_cursor` ? when set to 2, it adds a short error text as a virtual text.

**Comment by klausman** [comment]
Ah, yes, that seems to be the right setting (0 for me, 1 seems to still add comments for errors, 2 for errors and warnings).

It might be useful to update the docs to include some language about how this will usually be a comment in the used language, so people like me who search for "comment" and few other strings can find the text. I definitely would not expect comments to be be added to my code from the option's description here: https://github.com/dense-analysis/ale/blob/a18472cc58b85ba4e037fe3ad4c3f525aef74a85/doc/ale.txt#L2291

Either way, thanks a bunch!

**Reply by RyanSquared** [reply]
Out of curiosity, which LSP were you using that adds comment style virtualtexts? Whenever I've used it, it's never been a comment, just a bit of highlighted text after the code.

**Comment by klausman** [comment]
What is an LSP in this context?

FWIW, I was editing Go code using vim 8.

**Reply by RyanSquared** [reply]
LSP is Language Server Protocol. This would be the program that communicates between your editor and your codebase. I assume for this it would be `gopls` ? You can check which linters (including LSPs) are enabled by running `:ALEInfo`. I don't use Go so I can't advise further on why it would show up as a comment.

**Comment by klausman** [comment]
So with the default setting, a spurious `!` results in this:

![image](https://user-images.githubusercontent.com/4527946/209987508-2e65eee1-45ce-454c-8046-868e2dc0a0c4.png)

whereas in older versions (and with g:ale_virtualtext_cursor=0) it looks like this:

![image](https://user-images.githubusercontent.com/4527946/209987575-7cec5217-453f-406e-ab6b-01414be2450f.png)

Note how the latter is much cleaner/less distracting, but yet conveys the same information. Fortunately, writing the file will not include the comments added. I guess in some contexts, adding the comment allows for more useful information or something?

Anyway, I can have my old statusline behaviour back, and that's what I care about :)

**Reply by w0rp** [reply]
I want to maintain it so you never in all of history have to change that setting again too.

I purposefully made the default annoying as Hell, and I'll stick with it, as my philosophy is that the default for ALE is that it says "FIX. YOUR. BROKEN. CODE." ... In so many words at least.

I'm also going to generally clean up what ALE prints as linters return a lot of garbage which actually cause a general performace problem. There are worst offenders in Python or C where a single wrong bit of syntax causes linters to say, "So, this problem on line 7? I'm gonna make that 200 errors in your file." This is basically garbage output, and I plan on throwing away garbage output in Vim script wherever possible, which will fix the performance problem caused by dealing with that many errors, and make the virtual text default less annoying. See here: https://github.com/dense-analysis/ale/issues/4208
